folsom-elasticache/src/main/java/com/spotify/folsom/elasticache/ElastiCacheResolver.java

```diff
@@ -181,5 +196,13 @@ public Response resolve() {
         throw new RuntimeException("ElastiCache auto-discovery failed", e);
       }
     }
+
+    private Socket createSocket() throws IOException {
+      if (useTls) {
+        return SSLSocketFactory.getDefault().createSocket();
```

Comment:
This is a bit opionated but given that the purpose of this resolver is to connect to AWS resource alternative methods won't be required anyway.
I checked that on a real application and it works perfectly fine.